### PR TITLE
Backports 0.18 pr4

### PR DESCRIPTION
**Pull request by Munkybooty**

### CODE REVIEW

File: depends/config.site.in [hunk 1]
```diff
@@ -55,9 +55,12 @@ export PKG_CONFIG_PATH=$depends_prefix/share/pkgconfig
 CPPFLAGS="-I$depends_prefix/include/ $CPPFLAGS"
 LDFLAGS="-L$depends_prefix/lib $LDFLAGS"
 
-CC="@CC@"
-CXX="@CXX@"
-OBJC="${CC}"
```
Comment (PastaPastaPasta) [inline depends/config.site.in:60]: at best 14036 should be marked partial.

This doesn't actually backport the logic of running `unit tests --with-sanitizers=undefined` in CI

File: contrib/testgen/gen_key_io_test_vectors.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    PYTHONPATH=../../test/functional/test_framework ./gen_key_io_test_vectors.py invalid 50 > ../../src/test/data/key_io_invalid.json
+'''
+# 2012 Wladimir J. van der Laan
+# Released under MIT License
+import os
+from itertools import islice
+from base58 import b58encode_chk, b58decode_chk, b58chars
+import random
+from binascii import b2a_hex
+from segwit_addr import bech32_encode, decode, convertbits, CHARSET
```
Comment (PastaPastaPasta) [inline contrib/testgen/gen_key_io_test_vectors.py:19]: 13935 I doubt this script actually works

File: .travis.yml [hunk 3]
```diff
@@ -15,7 +15,6 @@ cache:
   ccache: true
   directories:
     - $HOME/cache
-
 env:
```
Comment (PastaPastaPasta) [inline .travis.yml:18]: 13863 our travis.yml likely needs some more changes...

Comment (Munkybooty) [reply .travis.yml:18]: should be fixed up now

**Comment by UdjinM6** [comment]
Should rebase and drop all already merged PRs.

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
Pls see 00988b1fa638aef50131c84456adad41509d9365, 70b47f22ab05da2bcde7c89a5f06c76946e96ae0 and 00748caa3624feaed3a7a51066c3a71ea878e936 (enabling sanitizers in ci build causes memory exhaustion though which can be fixed like that ffc51b9982e880f8de98ba75b8ecebd141c1b892). Also, we don't really use travis anymore and we have our own scripts for CI so 13683 should be smth like 77bbcd96122ea34dd2875204d90f3c87a39759d7 (14081 should be dropped/skipped).

**Comment by UdjinM6** [comment]
You merged ffc51b9 into the wrong PR - it was fixing ci builds after 14036, not 13863. I would probably even keep that fix as a separate commit, not sure. LGTM otherwise.

**Comment by Munkybooty** [comment]
> You merged [ffc51b9](https://github.com/dashpay/dash/commit/ffc51b9982e880f8de98ba75b8ecebd141c1b892) into the wrong PR - it was fixing ci builds after 14036, not 13863. I would probably even keep that fix as a separate commit, not sure. LGTM otherwise.

Wow, i can't believe i actually messed that up. I'll look into fixing that.

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**